test/scripts/kind_provisioner.sh

```diff
@@ -9,22 +9,33 @@ set -x
 
 ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
 
-source "$ROOT/scripts/lib.sh"
 source "$ROOT/scripts/cluster.sh"
+source "$ROOT/scripts/lib.sh"
 
 clusters=("east" "west")
 all_clusters=$(kind get clusters 2>&1)
 matching_clusters=$(echo "$all_clusters" | grep -c -E "$(printf '%s|' "${clusters[@]}" | sed 's/|$//')" || true)
 
 if [ "$matching_clusters" -eq 0 ]; then
   echo "No required clusters found. Provisioning..."
-  provision_kind_clusters
+
+  pids=()
+  for cluster in "${clusters[@]}"; do
+    provision_kind_cluster "${cluster}" &
+    pids+=($!)
+  done
+
+  for pid in "${pids[@]}"; do
+    wait "$pid" || echo "Process $pid failed"
+  done
```

Comment:
Should this exit with a non-zero status if the process fails?

Reply:
Yeah, we could. If any of the cluster provisioning fails, we need to recreate the whole setup regardless, based on the current logic that this refactoring preserved.

Fixed in a7594b4